### PR TITLE
[devops] fix ci for document check

### LINKED ISSUE #3753: [devops] fix ci for document check
## Overview

https://github.com/hpcaitech/ColossalAI/actions/runs/4977382383/jobs/8906746190?pr=3729#step:5:14

This shows that it failed to update the main branch when merging. We expect that the main branch should be the merged (latest) branch. But now, the main branch is origin main branch.

We can directly checkout main branch from curBranch, so that the main branch is definitely the merged (latest) branch.

## Goal
1. Fix branch merge behavior.
2. Update comments.

### PR DESCRIPTION
**Pull request by ver217**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

Closes #3753 

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

As described in #3753 , the merge behavior is unexpected, in this PR:

1. Fix branch merge behavior.
2. Update comments.

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: .github/workflows/doc_check_on_pr.yml [hunk 1]
```diff
@@ -50,9 +50,11 @@ jobs:
       - name: Make the merged branch main
         run: |
           cd ColossalAI
-          curBranch=$(git rev-parse --abbrev-ref HEAD)
-          git checkout main
-          git merge $curBranch             # fast-forward master up to the merge
+          git checkout -b main
+          git branch -u origin/main
+          git config user.name 'github-actions'
+          git config user.email 'github-actions@github.com'
```
Comment (FrankLeeeee) [inline .github/workflows/doc_check_on_pr.yml:56]: I think the reason why the current merged branch needs to be set to to main is that the documentation will treat the main branch as the `current` version. The whole workflow is like this:

1. the github actions `checkout` will merge the source branch and the target branch, this branch can be obtained from `$(git rev-parse --abbrev-ref HEAD)`, let's name this branch as `curBranch`
2. make this `curBranch` the main branch
3. documentation build will then take the main branch to build documentation

Not sure whether anything wrong with the merge?
5.

Comment (ver217) [reply .github/workflows/doc_check_on_pr.yml:56]: As https://github.com/hpcaitech/ColossalAI/actions/runs/4977382383/jobs/8906746190?pr=3729#step:5:14, it shows `Already up to date.` when merging. So no change is applied.

Comment (ver217) [reply .github/workflows/doc_check_on_pr.yml:56]: We can directly use the merged branch (`curBranch`) as the main branch.

Comment (FrankLeeeee) [reply .github/workflows/doc_check_on_pr.yml:58]: Nice!

**Review by FrankLeeeee — APPROVED** [review]

**Pull request merged**